pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -186,39 +202,68 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v, certVersion %v, vaultURI: %v", keyVaultCert.Name, keyVaultCert.Version, s.vaultURI)
 
 		startTime := time.Now()
-		secretResponse, err := s.secretKVClient.GetSecret(ctx, keyVaultCert.Name, keyVaultCert.Version)
-		if err != nil {
-			if isSecretDisabledError(err) {
-				// if secret is disabled, get the version of the certificate for status
-				certResponse, err := s.certificateKVClient.GetCertificate(ctx, keyVaultCert.Name, keyVaultCert.Version)
-				if err != nil {
-					return nil, nil, fmt.Errorf("failed to get certificate objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
-				}
-				certBundle := certResponse.CertificateBundle
-				keyVaultCert.Version = getObjectVersion(*certBundle.KID)
-				isEnabled := *certBundle.Attributes.Enabled
-				lastRefreshed := startTime.Format(time.RFC3339)
-				certProperty := getStatusProperty(keyVaultCert.Name, keyVaultCert.Version, lastRefreshed, isEnabled)
-				certsStatus = append(certsStatus, certProperty)
-				mapKey := keymanagementprovider.KMPMapKey{Name: keyVaultCert.Name, Version: keyVaultCert.Version, Enabled: isEnabled}
-				keymanagementprovider.DeleteCertificateFromMap(s.resource, mapKey)
-				continue
+		// if versionHistoryLimit is not set, set it to default value 1
+		if keyVaultCert.VersionHistoryLimit == 0 {
+			keyVaultCert.VersionHistoryLimit = versionHistoryLimitDefault
+		}
+
+		var versionHistory []struct {
+			Version string
+			Created time.Time
+		}
+
+		certVersionPager := s.certificateKVClient.NewListCertificateVersionsPager(keyVaultCert.Name, nil)
+		for certVersionPager.More() {
+			pager, err := certVersionPager.NextPage(ctx)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to get certificate versions for objectName:%s, error: %w", keyVaultCert.Name, err)
+			}
+			for _, cert := range pager.Value {
+				versionHistory = append(versionHistory, struct {
+					Version string
+					Created time.Time
+				}{Version: cert.ID.Version(), Created: *cert.Attributes.Created})
 			}
-			return nil, nil, fmt.Errorf("failed to get secret objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
 		}
 
-		secretBundle := secretResponse.SecretBundle
-		isEnabled := *secretBundle.Attributes.Enabled
+		// sort the version history by created time
+		sortVersionHistory(versionHistory)
```

Comment:
can you leave a comment in code explaining why a separate method is needed called `GetSortedVersions`?